### ISSUE TITLE
Help Message is Displayed with Incorrect Subcommand

### ISSUE THREAD

**Issue by danielhelfand** [knative/client#884]
Typing an incorrect subcommand results in the entire help message being displayed. An example is shown below:

```
$ kn service l
Service command group

Usage:
  kn service [flags]
  kn service [command]

Available Commands:
  create      Create a service.
  delete      Delete a service.
  describe    Show details of a service
  export      Export a service.
  list        List available services.
  update      Update a service.

Flags:
  -h, --help   help for service

Global Flags:
      --config string       kn config file (default is ~/.config/kn/config.yaml)
      --kubeconfig string   kubectl config file (default is ~/.kube/config)
      --log-http            log http traffic

Use "kn service [command] --help" for more information about a command.
unknown sub-command "l" for "kn service"
```

This makes finding the true error message difficult to find, and is similar to displaying usage with an error message. In most cases, this error will most likely come from a typo, and it might be more helpful to suggest using `--help` as opposed to displaying the entire `--help` message for the command by default:

`unknown sub-command "l" for "kn service". Run kn service --help for usage.`
This could be accomplished by removing this line in [`root.go`](https://github.com/knative/client/blob/ff350fb6a70f4903e8744654fa2e470ea048811d/pkg/kn/core/root.go#L223).

**Comment by rhuss** [comment]
I agree. If there are not objection I'm going to add this to #877 , too. But if one want to need this quicker I think its just a matter of remove https://github.com/knative/client/blob/834ee7904932a43b1ab4ef0cdc0ce6a038fda7e4/pkg/kn/core/root.go#L223

**Comment by rhuss** [comment]
ah, sorry. Should read the full description of the issue, you found already that spot. ;-)

**Comment by rhuss** [comment]
Fixed with #877 being merged.
